controller/workitem.go

```diff
@@ -136,6 +142,23 @@ func (c *WorkitemController) Update(ctx *app.UpdateWorkitemContext) error {
 	if err != nil {
 		return jsonapi.JSONErrorResponse(ctx, errs.Wrapf(err, "failed to load work item type: %s", wi.Type))
 	}
+	// if this WIs WIT has an action rule defined, run it.
+	if wit.TransRuleKey != "" {
+		// first, create the change set.
+		changes, err := wi.ChangeSet(*oldWi)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, errs.Wrapf(err, "failed to create work item changeset"))
+		}
+		// then execute the action, overwriting the existing wi as the action might have a sideffect on the wi.
+		newContext, _, err := actions.ExecuteActionsByChangeset(ctx, c.db, *currentUserIdentityID, *wi, changes, map[string]string{
+			wit.TransRuleKey: wit.TransRuleArgument,
+		})
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, errs.Wrapf(err, "failed to execute update actions on work item"))
+		}
+		result := newContext.(workitem.WorkItem)
+		wi = &result
```

Comment:
We already did save the work item above when we reach this point. It seems dangerous to assume that this overwrite doesn't do any harm, so instead of overwriting I suggest to load the work item from the DB again and convert it. Then we're safe that nothing was messed up.

Reply:
I don't understand this comment. `newContext` is the WI after the action execution. The action itself stores it to the database. So `newContext` always holds the latest version.

Reply:
@michaelkleinhenz I agree with @kwk , safer to load it from the DB again rather than making an assumption.